lib/fauna/page.rb

```diff
@@ -0,0 +1,381 @@
+  #     map(page) { |ref| select ['data', 'value'], get(ref) }
+  #   end
+  #
+  #   # Same thing, but using builders instead
+  #
+  #   page = Page.new(client, Query.match(Ref('indexes/items'))).with_size(5).with_fauna_map do |page|
+  #     map(page) { |ref| select ['data', 'value'], get(ref) }
+  #   end
+  class Page
+    ##
+    # Creates a pagination helper for paging/iterating over a set.
+    #
+    # +client+:: Client to execute queries with.
+    # +set+:: A set query to paginate over.
+    # +params+:: A list of parameters to pass to {paginate}[https://faunadb.com/documentation/queries#read_functions-paginate_set].
+    # +fauna_map+:: Optional block to wrap the generated paginate query with. The block will be run in a query context.
+    #               The paginate query will be passed into the block as an argument.
+    def initialize(client, set, params = {}, &fauna_map)
+      @client = client
+      @set = set
+      @page_params = params
+      @fauna_map = fauna_map
+      @ruby_map = nil
+
+      unload_page
+    end
+
+    # Returns +true+ if +other+ is a Page and contains the same configuration and data.
+    def ==(other)
+      return false unless other.is_a? Page
+      data == other.data && before == other.before && after == other.after &&
+        populated == other.populated && client == other.client && set == other.set &&
+        @page_params == other.instance_variable_get(:@page_params) &&
+        fauna_map == other.fauna_map && ruby_map == other.ruby_map
+    end
+
+    alias_method :eql?, :==
+
+    # :section: Configuration
+
+    ##
+    # If the current page is populated.
+    #
+    # Unpopulated pages will not auto-populate; to populate the page, begin paging in any direction.
+    attr_reader :populated
+
+    # Client to execute queries with.
+    attr_reader :client
+
+    # The set query to paginate over.
+    attr_reader :set
+
+    # The configured ts to use for pagination.
+    def ts
+      @page_params[:ts]
+    end
+
+    ##
+    # The configured cursor to use for pagination.
+    #
+    # Automatically updated during pagination. The appropriate cursor is used based on paging direction.
+    def cursor
+      @page_params.select { |key, _| CURSOR_KEYS.include? key }
+    end
+
+    # The configured size to use for pagination.
+    def size
+      @page_params[:size]
+    end
+
+    # If pagination will include events.
+    def events
+      @page_params[:events]
+    end
+
+    # If pagination will include sources.
+    def sources
+      @page_params[:sources]
+    end
+
+    # An optional block used to apply a fauna map to the generated pagination query.
+    attr_reader :fauna_map
+
+    # An optional block used to map the data returned from the fauna query.
+    attr_reader :ruby_map
+
+    # :section: Data
+
+    ##
+    # Data contained within the current page.
+    #
+    # Always +nil+ for the initial page. Call one of the pagination methods to begin paging.
```

Comment:
There's a mismatch then. Perhaps the first page just needs to be loaded, or if it's lazy, then appear as such.